File: haskell-geometry/src/Caculate.hs
module Caculate (distance) where

import Geometry (Geometry (..))
import Point (Point2D (..))

instance Geometry Point2D where
  distance pointA pointB =
    sqrt ((x pointA - x pointB) ^ 2 + (y pointA - y pointB) ^ 2)

  scale point factor =
    point
      { x = x point * factor,
        y = y point * factor
      }

File: go-geometry/caculate/caculate.go
package caculate

import (
	"gogeometry/point"
	"math"
)

type Point2DGeometry struct {
	point.Point2D
}

func (p *Point2DGeometry) Distance(b interface{}) float64 {
	pointA := p
	pointB := b.(*Point2DGeometry)

	return math.Sqrt(math.Pow(pointA.X - pointB.X, 2) + math.Pow(pointA.Y - pointB.Y, 2))
}

func (p *Point2DGeometry) Scale(factor float64) {
	p.X = p.X * 2
	p.Y = p.Y * 2
}
